docs/GETTING_STARTED.md

```diff
@@ -1,288 +1,247 @@
 
-```
+```toml
 [proxy_plugins]
   [proxy_plugins.soci]
     type = "snapshot"
     address = "/run/soci-snapshotter-grpc/soci-snapshotter-grpc.sock"
 ```
 
-These lines tell containerd that there is a snapshot plugin named SOCI and to
-communicate with it via a socket file.  Now let's restart containerd
+This config section tells containerd that there is a snapshot plugin named `soci`
+and to communicate with it via a socket file.
 
-```
+Now let's restart containerd and confirm containerd knows about soci-snapshotter plugin:
+
+```shell
 sudo systemctl restart containerd
+sudo ctr plugin ls
 ```
 
-To confirm that containerd knows about SOCI run
-
-```
-ctr plugin ls
-```
+`ctr plugin ls` lists all of the plugins from which you should see there is a
+`soci` plugin of type `io.containerd.snapshotter.v1`.
 
-This will list all of the plugins confirm that there is a **soci**
-plugin of type **io.containerd.snapshotter.v1**
+### Start soci-snapshotter
 
-### Preparing SOCI snapshotter
-First we need to start the snapshotter binary, it lives in the `out` directory.
-In the example command we arbitrarily are pushing the logs to a file named
-`~/soci-snapshotter-logs`, but please feel free to choose a different location:
+First we need to start the snapshotter grpc service binary (`soci-snapshotter-grpc`).
+Here we start the binary in background and simply redirect logs to an arbitrary file:
 
-```
-sudo ./soci-snapshotter-grpc&> ~/soci-snapshotter-logs &
+```shell
+sudo soci-snapshotter-grpc &> ~/soci-snapshotter-logs &
 ```
 
 Alternately, you can split up stdout (json logs) and stderr (plain text errors):
 
+```shell
+sudo soci-snapshotter-grpc 2> ~/soci-snapshotter-errors 1> ~/soci-snapshotter-logs &
 ```
-sudo ./soci-snapshotter-grpc 2> ~/soci-snapshotter-errors 1> ~/soci-snapshotter-logs &
-```
 
-Once the snapshotter is running we can call the rpull command from the SOCI CLI.
+### Lazily pull image
+
+Once the snapshotter is running we can call the `rpull` command from SOCI CLI.
 This command reads the manfiest from the registry and mounts FUSE filesystems
 for each layer.
 
-In this command the `--soci-index-digest` needs to be the digest of the SOCI
-index manifest.  The final argument is the tag of the image:
-
-**NOTE: Specifying `--soci-index-digest` is optional. If the argument is not provided, the snapshotter will use the [Referrers API](https://github.com/opencontainers/distribution-spec/blob/main/spec.md#listing-referrers) to fetch a list of available indices.**
-```
-sudo ./soci image rpull --plain-http --soci-index-digest sha256:f5f2a8558d0036c0a316638c5575607c01d1fa1588dbe56c6a5a7253e30ce107 localhost:5000/rabbitmq:latest
-```
+> The optional flag `--soci-index-digest` needs to be the digest of the SOCI index manifest.
+> If not provided, the snapshotter will use the [Referrers API](https://github.com/opencontainers/distribution-spec/blob/main/spec.md#listing-referrers) to fetch a list of available indices.
```

Comment:
I think this is confusing to people because they will either a) not know what the referrers API is, or b) know that their registry of choice doesn't support it. 

I think we should say something along the lines of:
"... use the OCI distribution-spec's Referrers API if available, otherwise the spec's fallback mechanism"

Reply:
added fallback mechanism with its link.